syncer/syncer.go

```diff
@@ -1554,6 +1572,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 			err2            error
 			exitSafeModeLoc binlog.Location
 		)
+		// TODO: why use a sharding-resync variable savedGlobalLastLocation?
```

Comment:
To make safe more `safe`.
pos1 ... pos2 ... pos3
If DM resync between pos1 and pos2, I want safe mode open till pos3